### ISSUE TITLE
TOML 1.1.0

### ISSUE THREAD

**Issue by pradyunsg** [toml-lang/toml#928]
I like having issues for tracking releases, to have a single point for discussion.

**Comment by pradyunsg** [comment]
I think after #924, it'll be a good moment to cut an 1.1.0-rc1.

**Comment by arp242** [comment]
I think it would be nice to also update the toml-test to TOML 1.1; I added a test for `\e` some time ago, but that's about it 😅

There are two parts to this; the first is updating https://github.com/burntSushi/toml to support TOML 1.1, which is needed because toml-test uses it. This is just something I need to spend a bit of time on; I don't expect that will be too difficult.

The second part is actually adding tests for TOML 1.1, for both valid and invalid edge cases, and this is something people here can help out with. This also requires identifying some of the "invalid" tests that are now "valid"; I added a `-toml` flag to specify the TOML version, so we can keep it compatible with 1.0 and 1.1 (specified in [`version.go`](https://github.com/BurntSushi/toml-test/blob/master/version.go)).

At any rate, if people feel interested to spend some time on thinking of test cases and making a PR then that'd be appreciated.

**Comment by ChristianSi** [comment]
I've tried doing a little triage regarding all open issues and PR, to see whether and how urgently they should become part of 1.1.0. Here's the result:

Should definitively be included (in some way or other):

* [Clarify Unicode and UTF-8 references #929](https://github.com/toml-lang/toml/pull/929): this is really just a small wording change.
* [Relax comment parsing (per discussion on #567) #924 ](https://github.com/toml-lang/toml/pull/924): this incorporates the changes from #929 and so should be handled afterwards; it too is a fairly minor change, but it changes the substance of the spec, so @pradyunsg might conceivably reject it. But in any case it should be decided upon.

To be decided whether we include them:

* [Clarify sub-millisecond precision is allowed #805](https://github.com/toml-lang/toml/pull/805): it is just a tiny change to two examples; it makes to decide whether it goes in or is rejected before the next release.
* [clarify string descriptions #875](https://github.com/toml-lang/toml/pull/875): this too is a relatively small wording change, which might be accepted or rejected.
* [Define appropriate implementation behaviors for extreme values #538](https://github.com/toml-lang/toml/issues/538): it's unclear whether will result in any changes, but it probably makes sense to give it some though before the release.
* [Feature request: allow hex floatingpoint values #562](https://github.com/toml-lang/toml/issues/562): this would add new syntax, so we may accept it, postpone it for possibly inclusion in a future version, or reject it altogether.
* [Feature request: Add a duration/timedelta type #514](https://github.com/toml-lang/toml/issues/514) – related PR: [Proposal to add a duration type #717](https://github.com/toml-lang/toml/pull/717): this would add a new datatype, which might be accepted (in some way), postpones, and rejected.
* [Added Schema validator specification #116](https://github.com/toml-lang/toml/pull/116): this is pretty old, I don't know if it's still wanted or relevant. But it might make sense to decide on it some way or the other instead of just letting it rot.

Not essential, but would be nice to have:

* [Newlogo #909](https://github.com/toml-lang/toml/pull/909) – related issue: [New logo design concepts #828](https://github.com/toml-lang/toml/issues/828): this has been hanging for quite a while and for no good reason I can see; it's not really relevant for a new version, but would be good to act upon to improve TOML's corporate identity.
* [Maintainance and onboarding new maintainers #895](https://github.com/toml-lang/toml/issues/895): clearly we can do a new release with the existing maintainer team, but the maintainer bottleneck is still [noticeable](https://github.com/toml-lang/toml/pull/924#issuecomment-1292179217), so it would certainly be helpful if there was some substantial progress on this ticket.
* [Make toml-test the standard test suite for checking compliance of implementations #585](https://github.com/toml-lang/toml/issues/585): not really a part of the spec, but progress on this would be good as well.

To be started afterwards:

* [TOML as an RFC? #870](https://github.com/toml-lang/toml/issues/870): there was already some talk that the 1.1.0 release might be a good starting point for an RFC (if and when we decide we want want), so the release must be done first before this can be tackled.

New syntax or features that are probably beyond the scope of TOML 1.1 (if they ever
come):

* [Add Filesize #912](https://github.com/toml-lang/toml/issues/912): frankly I think we could just reject this; but if it comes, I don't think it'll come soon.
* [Proposal: TOML Schema #792](https://github.com/toml-lang/toml/issues/792): this is a big topic that's probably best treated separately
* [Proposal: Implementation-defined values #707](https://github.com/toml-lang/toml/issues/707): if this comes, it would be big topic that'll require a lot of work, so I don't see it as a possible part of 1.1. Frankly I also see it as a violation of TOML's "minimalism" principle and would therefore be in favor of closing it altogether.
* [Will custom type syntax be good for TOML health? #603](https://github.com/toml-lang/toml/issues/603): similar to #707 in scope and consequences.

Definitively beyond the scope of any TOML 1.x release, since it would be a breaking change:

* [Possible positional restrictions on table definitions #446](https://github.com/toml-lang/toml/issues/446): @pradyunsg: maybe one could add a "post-1.x" label for this, since it's incompatible with TOML 1.0 and so couldn't ever come before 2.0. Though frankly I think one could just close it, since "You can put tables in any order (except where arrays of tables are involved)" has always been a core element of TOML's philosophy and I don't really see it going away.

**Comment by marzer** [comment]
I really would like #562 to be included in the next version. We mandate that floats be IEEE754 binary 64, but we have no way of expressing those _exactly_ in a TOML document, and sometimes you need to do just that.

Of course an application can ask users to express this as a `[sign, exponent, mantissa]` triplet, but that's goofy.

**Comment by ChristianSi** [comment]
I'd also second @awvwgk's proposal to do a RC candidate first (once everything seems taken care of) – announce it widely, what for 2 months to see if anything comes up and if that's not the case (beyond typofixes and such), release 1.1 final after the 2 months. If not, release RC-2 and iterate. That's how we did it for 1.0, and it seems to have worked well enough.

**Comment by mcarans** [comment]
Now that what is under "Should definitively be included (in some way or other)" is merged, what is the process to decide for the ones under "To be decided whether we include them" in [this comment](https://github.com/toml-lang/toml/issues/928#issuecomment-1294659512)?

**Comment by mcarans** [comment]
> I think after #924, it'll be a good moment to cut an 1.1.0-rc1.
@pradyunsg How are things going towards the release candidate?

**Comment by pradyunsg** [comment]
> @pradyunsg How are things going towards the release candidate?

You can see this by looking at the milestone @mcarans -- https://github.com/toml-lang/toml/milestone/1.

**Comment by pradyunsg** [comment]
The way I've got things scoped out for 1.1.0 is that we'll have to resolve the unicode keys question and move forward with rc1. I don't plan to add any more new syntax/data type in TOML 1.1.0.

https://github.com/toml-lang/toml/milestone/1 has the issues that I intend to "definitely close out" on that topic, after which it's going to be the mechanics of cutting a new TOML release.

**Comment by pradyunsg** [comment]
Of course, immediately after I post that, I notice that there's also some disucussion about the changes to comment's permitted control characters so added that into the milestone as well.

**Comment by samuelcolvin** [comment]
Any update on the progress of of 1.1.0 being released? Similarly, is there anything we can do to help move things forward?

I'm blocked on something waiting for multiline table support, its release seems tantalisingly close...

**Comment by ChristianSi** [comment]
@samuelcolvin There's essentially only one thing that's blocking 1.1, and that's #895. Right now TOML has one (or two?) inactive maintainer and one barely active one, and so everything is stuck. @pradyunsg, the last remaining maintainer, now has all the necessary rights to appoint more maintainers, and proposals have been made, but so far he hasn't. So we're stuck.

Whether that'll ever change, and if so, when, only @pradyunsg could answer. Otherwise, your guess is as good as mine, or that of anyone else.

**Comment by paulehoffman** [comment]
Any news on this would be appreciated. I am hesitant to send developers to the main TOML site because it describes 1.0 and it doesn't even hint at a 1.1, which will have some significant changes to Unicode handling.

@pradyunsg: Is there anything blocking 1.1 that we can help on?

**Comment by pradyunsg** [comment]
> Is there anything blocking 1.1 that we can help on?

There was a bunch of discussion in #895 about this. At this point, the unblocking work has happened and there's mostly a couple of decisions + all the button clicking to do.

---

@arp242 Was there any particular reason you unpinned this issue? 👀 I put this back up since there wasn't a clear reason indicated for it.

**Comment by arp242** [comment]
> @arp242 Was there any particular reason you unpinned this issue? 👀 I put this back up since there wasn't a clear reason indicated for it.

Just didn't strike me as useful; but if it's helpful for folks then sure.

**Comment by 3052** [comment]
could we please even do a patch release? like 1.0.1? for goodness sake, important changes like newline for inline table have been merged for years already

https://github.com/toml-lang/toml/pull/904

**Comment by ChristianSi** [comment]
@3052 TOML 1.1 should get fairly close by now. I don't think there are any major hurdles left that prevent the publication of a release candidate.

**Comment by 3052** [comment]
@ChristianSi  ...

**Comment by mortie** [comment]
Out of curiosity, is there an actual, concrete planned time table? 1.1 has seemingly been "right around the corner" since 2022...

**Comment by ChristianSi** [comment]
@mortie There's no timeline, but as I said, the hurdles seem by now to be largely resolved, so it's just a question of someone leading is through the process towards publishing a release candidate. Maybe I'll manage within the next few days.

**Comment by ChristianSi** [comment]
@arp242 Sounds plausible! Though I'll still want to proofread that diff before we actually release it.

**Comment by nickeb96** [comment]
Is there a timeline on when this can be released?  It seems like everything has been completed

**Comment by Malix-Labs** [comment]
Can you push [its tag](https://github.com/toml-lang/toml/releases/tag/1.1.0) to a new [release](https://github.com/toml-lang/toml/releases) please?
